### PR TITLE
Improve the experiment client

### PR DESCRIPTION
**Pull request by tonyseek**
### Changes

- Refactoring the shortcut methods (`.get`, `.post` and also)
- Added cached feature for session clients (`OAuthState`)
- Added new method `make_client` to eliminate repeated code
- Fixed the specific token usage support: `client.get(url, token=foo)` instead of always calling `tokengetter`

@lepture Please review. Thanks.

### CODE REVIEW

**Comment by tonicbupt** [comment]
LGTM :beers:

**Comment by tonyseek** [comment]
@tonicbupt = =......

**Comment by tonicbupt** [comment]
@tonyseek teacher tony memeda :kissing_heart:

File: flask_oauthlib/contrib/client/application.py [hunk 1]
```diff
@@ -91,25 +125,29 @@ def authorized_response(self):
...
 
+    def request(self, method, url, token=None, *args, **kwargs):
+        if token is None:
+            client = self.client
+        else:
+            client = self._make_client_with_token(token)
+        url = urljoin(self.endpoint_url, url)
+        return getattr(client, method)(url, *args, **kwargs)
+
     # magic: generate methods which forward to self.client
```
Comment (lepture) [inline flask_oauthlib/contrib/client/application.py]: I like things without magic. I prefer a little repeating on http methods.

Comment (tonyseek) [inline flask_oauthlib/contrib/client/application.py]: done in 1b9ea59

**Comment by tonyseek** [comment]
@lepture How about now?

**Comment by lepture** [comment]
Merged in https://github.com/lepture/flask-oauthlib/commit/280661db1453e51cc9e4a34778b1fdcb3dc01ed4

**Pull request closed without merging**